ScorchApp/src/components/util/Searcher.vue

```diff
@@ -0,0 +1,200 @@
+                    <thead>
+                        <tr>
+                            <th v-for="(key, index) in objectKeys" :key="index" v-on:click="sortByForm(key)">
+                                <i class="fa" v-bind:class = "{'fa-sort-desc' : (key == sortKey && sortDirection == 'desc'),
+                                                                        'fa-sort-asc' : (key == sortKey && sortDirection == 'asc'),
+                                                                        'fa-sort' : (key != sortKey)}" 
+                                                                        aria-hidden="true"/>{{key}}
+                            </th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        <tr v-for="(row, srIndex) in searchResults.slice((currentPage-1) * limitPerPage, currentPage * limitPerPage)" :key="srIndex" v-on:click="selectRow(row)">
+                            <td v-for="(key, keyIndex) in objectKeys" :key="keyIndex">{{row[key]}}</td>
+                        </tr>
+                    </tbody>
+                </table>
+            </div>
+            <ul class="pagination">
+                <li class="page-item clickable" v-on:click="currentPage = Math.max(currentPage-1, 1)">
+                    <a class="page-link" >
+                        <span aria-hidden="true">&laquo;</span>
+                    </a>
+                </li>
+                <li class="page-item clickable" v-for="(page, index) in pagesToShow" :key="index" v-on:click="currentPage = page" v-bind:class ="{active : page == currentPage}">
+                    <a class="page-link" >{{page}}</a>
+                </li>
+                
+                <li class="page-item clickable" v-on:click="currentPage = Math.min(currentPage+1, numPages)">
+                    <a class="page-link" >
+                        <span aria-hidden="true">&raquo;</span>
+                    </a>
+                </li>
+            </ul>
+        </div>
+    </div>
+    
+</template>
+
+<script>
+    import filter from 'lodash/filter'
+    import orderBy from 'lodash/orderBy'
+    import sortBy from 'lodash/sortBy'
+
+export default {
+    name : 'searcher',
+    
+    data() {
+      return {
+        searchTerm : '',
+        sortKey : '',
+        sortDirection : 'asc',
+        objectKeys: [],
+        keysPopulated : false,
+        currentPage : 1
+      }
+    },
+    props: ['searchData', 'limitPerPage'],
+    methods: {
+        selectRow(row){
+            this.$emit('search-row-selected', row);
+        },
+        sortByForm(key){
+            if(this.sortKey === key){
+                this.sortDirection = this.sortDirection === 'desc' ? 'asc' : 'desc';
+            }
+            else{
+                this.sortKey = key;
+                this.sortDirection = 'asc';
+            }
+        },
+        populateObjectKeys(results){
```

Comment:
if we are passing in the data we want to be searched on. I think we can assume that whatever we passed in has a uniform definition of the object's keys. You probably don't need to foreach through the entire search data and only take the first row.

Reply:
Also, in the store you are already specifying what keys you are picking to be fed into this. Maybe just add another prop to this component for the keys that are passed in .